### DISCUSSION TITLE
[Thesis] remove libgdx

### DISCUSSION THREAD

**Discussion by cagix** [Programmiermethoden/Dungeon#1368] [Ideas]
libgdx is starting to piss me off. the configuration is weird and the visualisation isn't that nice and in the end we don't have that much of the library. is there a way to implement the essential functionalities on our own?

edit: (aus #576) 
> [gurkenlabs/litiengine](https://github.com/gurkenlabs/litiengine) ist ein 2d-java-framework auf der basis von java und awt(!) und ohne weitere abhängigkeiten. zusätzlich integriert es bereits ecs, level-/tile-funktionalitäten u.v.m. ... 
> 
> damit könnten wir die etwas undurchsichtigen abhängigkeiten in libgdx (insb. die plattformabhängigen teile des frameworks, die beim nächsten update evtl. wieder für bauchschmerzen sorgen) ablösen mit einem reinen java-basierten ansatz. 
> 
> vermutlich könnten wir auch unser eigenes framework deutlich verschlanken und uns auf die eigentlichen spielelemente fokussieren.

**Comment by AMatutat** [comment]
Guess we could build our own "framework" for this (maybe with Java2D), but is it worth it? 

>and in the end we don't have that much of the library.

Disagree on that one. libgdx gives so many options for the students to work with (3D wouldnt be possible without libgdx).

**Comment by cagix** [comment]
yeah, you're right. the library does offer some interesting features. but considering the problems it brings with it, i actually wonder if it's worth it ;)

**Comment by cagix** [comment]
@AMatutat It seems to me that we are in dire need of a new category / label: "**rant**" ...  :-)

**Comment by AMatutat** [comment]
Should i call it "rant"  or better "cagix hates the world" ?

**Comment by cagix** [comment]
> Should i call it "rant" or better "cagix hates the world" ?

as you see fit 😈  but "rant" might be better, just in case you also need this 👼

**Comment by cagix** [comment]
**clean-up**: @AMatutat @Tobias321 Brauchen wir das noch und sollte das in die neuen Repos verschoben werden, oder kann das zu?

Vorschlag: Wir schieben das in das neue core-Repo und können dann die Aktivitäten von @Tobias im Swing-Dungeon verlinken.

**Comment by AMatutat** [comment]
> Vorschlag: Wir schieben das in das neue core-Repo und können dann die Aktivitäten von @tobias im Swing-Dungeon verlinken.

mach das

**Comment by cagix** [comment]
see https://github.com/PM-Dungeon/pmdungeon-swing/issues/2 and https://github.com/PM-Dungeon/pmdungeon-swing/issues/11

**Comment by cagix** [comment]
siehe auch #576

**Comment by AMatutat** [comment]
In https://github.com/akullpp/awesome-java werden auch ein paar Java-Spiele-Frameworks aufgelistet. Könnte man sich in diesen Zuge auch anschauen

**Comment by cagix** [comment]
mit #2109 umarmen wir die dunkle seite. wo die api aus libgdx sinnvoll ist, sollten wir sie nutzen... das drumherumarbeiten dauert zu lange und verursacht zu viele schmerzen.
